### ISSUE TITLE
Error EPERM rename tmp-loader

### ISSUE THREAD

**Issue by ttrentham** [jspm/github#38]
I'm trying to install Aurelia Skeleton Navigation from github.

https://github.com/aurelia/skeleton-navigation/releases

I ran the jspm registry config github command and the test for my credentials (via a token) was successful. Prior to that, I got the warning about the rate limit.

When I run jspm install -y, it starts downloading and installing but eventually gets an error similar to the following:

err  Error: EPERM, rename 'C:\Users\ttrentham\.jspm\packages\github\aurelia\.tmp-loader@0.4.0'

I actually opened an issue with Aurelia  which they closed.

https://github.com/aurelia/skeleton-navigation/issues/65

I tried the last suggestion which was to execute it from my home instead of at work behind our firewall. I get the same issue.

This is on a Windows 8.1 machine. I'm running my command prompt as an administrator. I've got GitHub for Windows installed.

Any other help would be appreciated.

**Comment by guybedford** [comment]
@ttrentham it sounds like you may have a different permission for your local `git` to `jspm` itself. If you run jspm with higher priviledges does that help? Also can you run `jspm cc` without errors?

**Comment by subesokun** [comment]
The path `C:\Users\ttrentham.jspm\packages\github\aurelia.tmp-loader@0.4.0` with the `.` between `aurelia.tmp-loader` looks somehow suspicious to me. Shouldn't it have been `aurelia\tmp-loader` instead?

**Comment by ttrentham** [comment]
@guybedford I'm running command prompt with administrator privileges so shouldn't that mean permissions shouldn't be an issue?

Here's what I get from jspm cc

C:\source\prototypes\aurelia\skeleton-navigation-0.11.0>jspm cc               
ok   Loader file cache cleared.                                               
ok   Package cache cleared.                                                   
ok   github cache cleared.                                                    
ok   jspm cache cleared.                                                      
ok   npm cache cleared.                                                       
                                                                              
warn All caches cleared.                                                      
     Please post an issue if you suspect the cache isn't invalidating properly
.                                                                             
     jspm install -f is equivalent to running a cache clear for that specific 
package tree.                                                                 
                                                                              
C:\source\prototypes\aurelia\skeleton-navigation-0.11.0>                      

And running jspm install -y after running "jspm cc" worked! Thanks very much for the solution. Do you know the cause of my original error and why clearing the cache fixed it?

**Comment by guybedford** [comment]
I have no idea why the original issue happened - but it sounds like some sort of cache corruption as a result of halting the install when the github credentials failed.

I will leave this up here to track that, and will see if I can replicate in windows myself.

**Comment by ttrentham** [comment]
If it helps, I tried to to "jspm install -y" the first time and got the warnings about the rate limit. I then ran "jspm registry config github" and added a personal access token to my github account and retried "jspm install -y". That's when I started getting the error. I was following the instructions from Aurelia's Get Started page (although apparently not in the right order ;) ).

**Comment by scottwarnick** [comment]
in an cmd prompt running as an administrator
npm cache clear
jspm cc
jspm install -f -y
worked for me

**Comment by crisptrutski** [comment]
@ttrentham are you perhaps running an anti-virus? trying to figure out the root cause of the issue and saw this https://github.com/joyent/node/issues/3051

**Comment by crisptrutski** [comment]
Wrong link, meant this: https://github.com/Medium/phantomjs/pull/28

**Comment by crisptrutski** [comment]
@guybedford Tried to reproduce this in a Win 7 VM, without luck. Strong suspicion third party indexing involved.

**Comment by crisptrutski** [comment]
If we can figure out where these stacktraces get thrown from perhaps we can put in some retry tolerance.

**Comment by jnemeth** [comment]
I had same original issue, after running jspm cc, jspm install -y worked without a problem.

**Comment by crisptrutski** [comment]
@jnemeth @ttrentham @scottwarnick are any of you able to reproduce the issue reliably again, eg by removing the github creds?

Tried 3 times and no luck, thinking there's an environmental factor. Thinking that extra logging run on a machine that's reproducing the error will be the best way to figure this out, but tricky without a stacktrace.

**Comment by guybedford** [comment]
We need to be able to replicate this case to fix it. If anyone can provide answers to @crisptrutski's questions that would help a lot to work towards this.

**Comment by alexisargyris** [comment]
Had this problem for a week now on a win10 machine. Tried cleaning npm's cache, as well as the -f and -y flags with no luck. I just came across this issue, tried 'jspm cc' and... problem solved! Is the cause of this problem related to a very similar situation with npm (e.g. see [here](https://github.com/npm/npm/issues/10826))? The anti-virus that is mentioned both here and at npm's case as a possible culprit was not related in my case.
